system/security/securesocket/securesocket.cpp

```diff
@@ -1155,7 +1155,147 @@ const char* strtok__(const char* s, const char* d, StringBuffer& tok)
     return s;
 }
 
-class CSecureSocketContext : implements ISecureSocketContext, public CInterface
+using BIO_ptr = std::unique_ptr<BIO, decltype(&::BIO_free)>;
```

Comment:
The code I suggested:
```
struct BIO_deleter
{
    void operator() ( BIO * bio )
    {
       BIO_free( bio ); 
    }
};
using unique_bioptr = std::unique_ptr<BIO, BIO_deleter>;

```
avoids the need to pass BIO_free to the constructor of each BIO_ptr.

Reply:
ok, that's a nice approach.

We should keep these helper objects and functions together.
We already have existing helpers in system/security/cryptohelper/cryptocommon.hpp,
both for owned semantics of these ssl objects, and for openssl exception creation.

I'd be happy for those to be switched to using std::unique_ptr in this PR, or it might be best to use them as is for now, and switch their implementation to std::unique_ptr at a later date.